llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -14945,6 +14945,13 @@ SDValue SITargetLowering::performMinMaxCombine(SDNode *N,
     }
   }
 
+  // max(x, neg(x)) -> abs(x)
```

Comment:
Can we do this in the generic part?

Reply:
> Can we do this in the generic part?

All ISD::SMAX nodes are sent to this function from the switch in PerformDAGCombine.  I don't see where else to put it.  This function is intended to process min or max nodes so it makes sense to put it here.

Reply:
I don't see any AMDGPU specific ISD here so why can't this be done in the generic combine part?